webdriver/tests/bidi/browsing_context/navigate/fragment_navigated.py

```diff
@@ -0,0 +1,101 @@
+import pytest
+
+from tests.support.sync import AsyncPoll
+from webdriver.bidi.modules.script import ContextTarget
+
+from ... import any_int, any_string, recursive_compare
+
+pytestmark = pytest.mark.asyncio
+
+EMPTY_PAGE = "/webdriver/tests/bidi/browsing_context/navigate/support/empty.html"
+FRAGMENT_NAVIGATED_EVENT = "browsingContext.fragmentNavigated"
+
+
+async def test_fragment_navigated_new_document(
```

Comment:
Mind following the style of tests that we already have for events? Have a look at the `domContentLoaded` tests as example. Those give a great base for what needs to be tested in general.

Reply:
Could you be more specific about what you mean by this (`./wpt lint` passes so I assume all style considerations are passing)? or do you mean test_unsubscribe/test_subscribe? I don't think it's worthwhile testing this for every event as those should be handled by the session module?

Reply:
yeah it's not about linting sorry. But what they test. Also the subscription and unsubscription we need to test that for each event given that each module has to run special steps and implementations might not behave correctly. Within the session module we test all the different argument combinations.

Reply:
do we want to test relative order of events? E.g., that fragmentNavigated is emitted before domContentLoaded and after contextCreated? Would it make sense to group event tests by the operation they are triggered by?

Reply:
added iframe and new context tests. I think that should cover the main cases?